constraint-solver/src/range_constraint.rs

```diff
@@ -744,6 +744,15 @@ mod test {
         assert!(d.allows_value(25.into()));
     }
 
+    #[test]
+    fn disjunction_mask() {
+        type F = GoldilocksField;
+        let a = RangeConstraint::<F>::from_mask(2u32);
+        let b = RangeConstraint::<F>::from_mask(1u32);
+        let d = a.disjunction(&b);
+        assert_eq!(d, RangeConstraint::from_mask(3u32));
```

Comment:
I don't think this is a bug. RC `a` only allows the two values `0` and `2`, while `b` only allows `0` and `1`. The disjunction (union of the ranges) would be `0, 1, 2`. Its mask is `3` but it's value range is `0..=2`. If you construct an RC from mask 3 directly, it also allows the value `3`, which is not the case for the disjunction.

Reply:
Right! I was only thinking in terms of masks, but you're right. We want a RC whose set of allowed values are the union of the input allowed values, and that's what we get.